src/ast/query.rs

```diff
@@ -455,6 +460,44 @@ impl fmt::Display for TableFactor {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct TableSample {
+    pub sampling_method: SamplingMethod,
+    pub sampling_fraction: Box<Expr>,
```

Comment:
SQL Syntax allows for the sampling_fraction and seed value to be a real valued expression but not arbitrary columns. is Expr still the best type for this member?

Reply:
@ayushdg @andygrove as the standard states ([1](https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#sample-clause)), we can have a huge amount of expressions there (described as a term ([2](https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#term)) and a few additions to it). It can support values, but a lot of additional things (e.g., numeric value functions ([3](https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#numeric-value-function))). Is it really reasonable to have a `Value` there?

Maybe have a task for an expansion to all other possible values, that includes but is not restricted to simple values?

[1] : https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#sample-clause
[2] : https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#term
[3] : https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#numeric-value-function